core/src/main/java/org/apache/iceberg/avro/AvroWithPartnerByStructureVisitor.java

```diff
@@ -51,7 +51,12 @@
         Preconditions.checkArgument(
             visitor.isStringType(keyType),
             "Invalid map: %s is not a string", keyType);
-        return visitor.map(partner, schema, visit(visitor.mapValueType(partner), schema.getValueType(), visitor));
+
+        visitor.beforeMapValue("value", schema.getValueType(), schema);
```

Comment:
Should this make an artificial call to the map key callbacks using a generic String schema as well?